### PR TITLE
Allow session protocol to be specified for HttpHealthCheckedEndpointG…

### PR DESCRIPTION
**Pull request by anuraaga**
…roup.

As there are many parameters now, added a builder instead of extending the static factories.

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupBuilder.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     */
+    public HttpHealthCheckedEndpointGroupBuilder(EndpointGroup delegate, String healthCheckPath) {
+        this.delegate = requireNonNull(delegate, "delegate");
+        this.healthCheckPath = requireNonNull(healthCheckPath, "healthCheckPath");
+    }
+
+    /**
+     * Sets the {@link SessionProtocol} to be used when making health check requests.
+     */
+    public HttpHealthCheckedEndpointGroupBuilder healthCheckProtocol(SessionProtocol healthCheckProtocol) {
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupBuilder.java]: Could remove `healthCheck` prefix?

File: core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupBuilder.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        this.healthCheckProtocol = requireNonNull(healthCheckProtocol, "healthCheckProtocol");
+        return this;
+    }
+
+    /**
+     * Sets the interval between health check requests. Must be positive.
+     */
+    public HttpHealthCheckedEndpointGroupBuilder healthCheckRetryInterval(Duration healthCheckRetryInterval) {
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupBuilder.java]: Could remove `healthCheck` prefix?

File: core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupBuilder.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        requireNonNull(healthCheckRetryInterval, "healthCheckRetryInterval");
+        checkArgument(!healthCheckRetryInterval.isNegative() && !healthCheckRetryInterval.isZero(),
+                      "healthCheckRetryInterval must be positive.");
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupBuilder.java]: `"retryInterval: %s (expected > 0)", retryInterval`?

File: core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupBuilder.java (hunk 1 above)
Comment (anuraaga) [reply core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupBuilder.java]: Good idea, updated

File: testing/src/main/java/com/linecorp/armeria/testing/server/ServerRule.java [hunk 4]
```diff
@@ -156,10 +153,20 @@ public int httpPort() {
...
      */
     public int httpsPort() {
+        return port(HTTPS);
+    }
+
+    /**
+     * Returns the port number of the {@link Server} for the specified {@link SessionProtocol}.
+     *
+     * @throws IllegalStateException if the {@link Server} is not started or it did not open a port of the
+     *     specified protocol.
```
Comment (minwoox) [inline testing/src/main/java/com/linecorp/armeria/testing/server/ServerRule.java]: indentation?

Comment (anuraaga) [reply testing/src/main/java/com/linecorp/armeria/testing/server/ServerRule.java]: This is the indentation checkstyle told me to use.

Comment (trustin) [reply testing/src/main/java/com/linecorp/armeria/testing/server/ServerRule.java]: Could look better if indentet at the same column with `if the..`

Comment (anuraaga) [reply testing/src/main/java/com/linecorp/armeria/testing/server/ServerRule.java]: Ah didn't realize the checkstyle was at least 4, not exactly 4 - fixed

**Comment by trustin** [comment]
Thanks, @anuraaga !

**Pull request merged**